controllers/token_scorer/token_scorer.py

```diff
@@ -0,0 +1,257 @@
+import sys
+import struct
+from math import cos, sin, sqrt, atan2
+from typing import cast, Dict, List, Tuple, NamedTuple
+from pathlib import Path
+
+# Webots specific library
+from controller import Receiver, Supervisor
+
+# Root directory of the SR webots simulator (equivalent to the root of the git repo)
+REPO_ROOT = Path(__file__).resolve().parent.parent.parent
+
+sys.path.insert(1, str(REPO_ROOT / 'modules'))
+
+from sr.robot.utils import get_robot_device  # isort:skip
+from shared_utils import Owner, Token, TOKENS  # isort:skip
+import controller_utils  # isort:skip
+
+SCORE_UPDATES_PER_SECOND = 10
+
+FLOOR_TOKEN_VALUE = 1
+PLATFORM_EXTRA_TOKEN_VALUE = 1
+
+
+class Point(NamedTuple):
+    # We are only tracking tokens on a 2D plane, with Y as north
```

Comment:
How does this align with the ruling around what defines a token as "in"?

Reply:
The intention was this would become the definition of a token being "in" since this will put tokens in the region with which their centers lie.

Reply:
I'm perfectly happy with a redefinition of "in" to match this implementation
It only needs to be unambiguous and easy to communicate to competitors. "Location of centre of token" is suitable for that